File: backends/gin/model/capsule.go
// Package model GORM 数据模型
package model

// Capsule 时间胶囊实体
type Capsule struct {
	ID        int64     `gorm:"primaryKey;autoIncrement"`
	Code      string    `gorm:"type:varchar(8);uniqueIndex;not null"`
	Title     string    `gorm:"type:varchar(100);not null"`
	Content   string    `gorm:"type:text;not null"`
	Creator   string    `gorm:"type:varchar(30);not null"`
	OpenAt    string    `gorm:"column:open_at;type:text;not null"`
	CreatedAt string    `gorm:"column:created_at;type:text;not null"`
}

// TableName 指定表名
func (Capsule) TableName() string {
	return "capsules"
}

File: fullstacks/laravel/app/Models/Capsule.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Capsule extends Model
{
    protected $table = 'capsules';
    protected $primaryKey = 'code';
    public $incrementing = false;
    protected $keyType = 'string';
    public $timestamps = false;

    protected $fillable = [
        'code', 'title', 'content', 'creator', 'open_at', 'created_at',
    ];

    protected function casts(): array
    {
        return [
            'open_at' => 'datetime',
            'created_at' => 'datetime',
        ];
    }

    public function isOpened(): bool
    {
        return now()->greaterThanOrEqualTo($this->open_at);
    }
}
